qiskit_experiments/analysis/data_processing.py

```diff
@@ -80,44 +85,43 @@ def mean_xy_data(
     Raises:
         QiskitError: if "ivw" method is used without providing a sigma.
     """
-    x_means = np.unique(xdata, axis=0)
-    y_means = np.zeros(x_means.size)
-    y_sigmas = np.zeros(x_means.size)
-
-    # Sample mean and variance method
-    if method == "sample":
-        for i in range(x_means.size):
-            # Get positions of y to average
-            idxs = xdata == x_means[i]
-            ys = ydata[idxs]
-
+    if method == "iwv" and sigma is None:
+        raise QiskitError(
+            "The inverse-weighted variance method cannot be used with" " `sigma=None`"
+        )
+    if method not in ["sample", "iwv"]:
+        raise QiskitError(f"Unsupported method {method}")
+    xseries = series if series is not None else np.zeros(xdata.size)
+    x_keys = np.unique(np.column_stack((xdata, xseries)), axis=0)
+    x_means = np.zeros(x_keys.shape[0])
+    y_means = np.zeros(x_keys.shape[0])
+    y_sigmas = np.zeros(x_keys.shape[0])
+    series_mean = np.zeros(x_keys.shape[0])
+
+    for i, (x_val, xseries_val) in enumerate(x_keys):
+        x_means[i] = x_val
+        series_mean[i] = xseries_val
+        # Get positions of y to average
+        idxs = np.where((xdata == x_val) & (xseries == xseries_val))
+        ys = ydata[idxs]
+
+        # Sample mean and variance method
+        if method == "sample":
             # Compute sample mean and biased sample variance
             y_means[i] = np.mean(ys)
             y_sigmas[i] = np.mean((y_means[i] - ys) ** 2)
+            # Inverse-weighted variance method
 
-        return x_means, y_means, y_sigmas
-
-    # Inverse-weighted variance method
-    if method == "iwv":
-        if sigma is None:
-            raise QiskitError(
-                "The inverse-weighted variance method cannot be used with" " `sigma=None`"
-            )
-        for i in range(x_means.size):
-            # Get positions of y to average
-            idxs = xdata == x_means[i]
-            ys = ydata[idxs]
-
+        if method == "iwv":
```

Comment:
elif

Reply:
Not sure why - the cases are mutually exclusive

Reply:
https://stackoverflow.com/questions/22782785/when-to-use-if-vs-elif-in-python